poketechnika/InputManager.cpp

```diff
@@ -28,6 +29,12 @@ bool InputManager::getAction(Action action) const {
 }
 
 void InputManager::processInput(sf::RenderWindow& window, const State& state) {
+	// Check for real-time actions
+	if (sf::Keyboard::isKeyPressed(sf::Keyboard::A)) _active_actions.insert(Action::MoveLeft);
+	if (sf::Keyboard::isKeyPressed(sf::Keyboard::D)) _active_actions.insert(Action::MoveRight);
+	if (sf::Keyboard::isKeyPressed(sf::Keyboard::W)) _active_actions.insert(Action::MoveUp);
+	if (sf::Keyboard::isKeyPressed(sf::Keyboard::S)) _active_actions.insert(Action::MoveDown);
+	// Check for event-type actions
```

Comment:
Rozumiem że Keyboard::isKeyPressed działa pod spodem inaczej niż Event::KeyPressed? Jeśli tak, to może niech wszystkie inputy z klawiatury będą ogarniane w ten sam sposób? Nie wiem czy potrzeba takiego rozróżnienia

Reply:
isKeyPressed działa bezpośrednio (sprawdzasz czy jest wciśnięte), natomiast Event::KeyPressed działa na zasadzie eventu, który musi zostać obsłużony. Zakomentuj sobie tą część i odpal, będzie wtedy śmigać na samych eventach i porównaj jak chodzi ludzik.

Imo można zostawić oba, Eventy bardziej się przydadzą podczas np. zmieniania menu czy przycisków

Reply:
Okej, ale nie pasuje mi trochę seria czterech ifów pod rząd. Można to rozwiązać podobnie jak eventy, osobną mapą

Reply:
spoko, to taki placeholder trochę żeby ogarnąć jak to działa